File: 1408.String Matching in an Array.rb
# @param {String[]} words
# @return {String[]}
def string_matching(words)
    h = Hash.new { |h, k| h[k] = [] }
    words.each { |word| h[word.size] << word}
    keys = h.keys.sort
    res = []
    (0...keys.size).each do |index|
        h[keys[index]].each do |word1|
            do_break = false
            ((index + 1)...keys.size).each do |index2|
                break if do_break
                h[keys[index2]].each do |word2|
                    break if do_break
                    if word2.include? word1
                        res << word1
                        do_break = true 
                    end
                end
            end
        end
    end
    res
end

File: 1001-1500/1408.String Matching in an Array.py
class Solution(object):
    def stringMatching(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        res = []
        for w1 in words:
            for w2 in words:
                if w1 != w2 and w1 in w2:
                    res.append(w1)
                    break
        return res
